Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/RetrieveProjectController.java

```diff
@@ -106,16 +106,30 @@ public ModelAndView getProjectList(Principal principal) throws QuadrigaStorageEx
                 }
             }
         }
-
-        // Fetch all the projects for which the user is associated workspace
-        // collaborator
+    
+        // Fetch all the projects for which the user is associated workspace collaborator.
         List<IProject> projectListAsWSCollaborator = projectManager.getProjectListAsWorkspaceCollaborator(userName);
+     
         if (projectListAsWSCollaborator != null) {
             for (IProject p : projectListAsWSCollaborator) {
+               // Process the project details if it has not been evaluated.
                 if (!projectIds.contains(p.getProjectId())) {
-                    fullProjects.add(projectManager.getProjectDetails(p.getProjectId()));
+                    // Get details of the project using the project id.
+                    IProject tempProject = projectManager.getProjectDetails(p.getProjectId());
+                    if(tempProject != null){
+                        List<IProjectWorkspace> tempProjectWorkspaces = new ArrayList<IProjectWorkspace>();
+                        // Examine the workspace details associated with the project. Include details of only those workspaces, where the user is a collaborator.
+                        for(IProjectWorkspace projectWorkspace : tempProject.getProjectWorkspaces()){
```

Comment:
Its strange that tempProject.getProjectWorkspaces() returns null while testing at your end. It could imply that workspace details were not set for the project. Could not replicate the issue at my end. Have modified the filter if(tempProject != null && tempProject.getProjectWorkspaces() != null)  to handle the any null pointer occurrence in the statement